### ISSUE TITLE
Clarify 'TimeCCdEnemyHeroes'

### ISSUE THREAD

**Issue by barrett777** [Blizzard/heroprotocol#21]
Could you clarify 'TimeCCdEnemyHeroes' in the 'ScoreResultEvent'?

First, is this value in seconds or frames?

Second, what exactly does this include?  It looks like it applies to Slows as well as Stuns.  Does it apply to Silences?

I have a replay with a 'Lunara' having a value of 90.  I don't think she has any stuns, so does this value include time a Hero is slowed?

I also see Murky with a value of 729, so I'm guessing the slow that his 'Q' applies must apply to this.

I see an Abathur with a value of 21 as well.  However, he did not take the 'Vile Nest' talent.  Do you know what may be applying to his value?

**Comment by pedrobaracho** [comment]
From my understanding slows apply. Stitches has AoE slow with his Putrid Bile that applies to a whole team, that is probably why you are getting a big number on him.

And from what I get on my replay files, I would say it is seconds, not frames. I dont know if they have a factor for slow either, for instance 50% slow being .5.

It would be nice to get an official comment on this.

**Comment by crorella** [comment]
yeah I think slows is considered, for example, Aba mines with the slow talent increases this value.
I think is in seconds * 4096.

**Comment by barrett777** [comment]
If it's seconds * 4096, that would mean Murky slowed players for 0.17 seconds with a value of 729.  Even if that only counts Octo-grab, it should be higher than that.  Same with Lunara having a value of 90, or Abathur with 21

**Comment by krypt0s** [comment]
Could the 21 seconds of Abathur (without Vile Nest) be from something used during Ultimate Evolution? I'm also seeing huge values for Stitches, almost always greatly exceeding the length of the match. The values are definitely in seconds and not seconds * 4096.

**Comment by crorella** [comment]
Sorry, my bad, just checked how I parsed it and I'm not applying any transformation on the value:

```
self.heroList[hero].secondsCCOnEnemies = int(instance['m_values'][hero][0]['m_value'])
```

**Comment by tenarsis** [comment]
TimeCCdEnemyHeroes is in game seconds written in fixed point format. So like you described, it's seconds * 4096.

It currently includes Slows, Stuns, Roots, and Stasis.

As to it exceeding the length of the match, that could occur due to the way it is tracked. Let's say you are both stunned and slowed for 5 seconds, that would report a 10 second CC.

Thanks for bringing this issue up.

**Comment by barrett777** [comment]
Thanks for the reply.  I do agree a 5 second slow on two heroes should count as 10 seconds in this data.

I'm not sure if I will really be able to use this number effectively if it includes slows and stuns - is there any way you could separate these into 4 different fields? :D

Stuns are crucial, and their detail would be lost in a broad field like this.  For example, I'd be very interested to see how long a Murky stunned people with Octo'grab, or as an example how efffective a Stitches is with the old AoE stun talent 'Pulverize'

**Comment by jwfergus** [comment]
I too would really like to see an explanation or fix of the values in here.

**Comment by barrett777** [comment]
There are still some issues here:

I have an Anub'arak showing over 18 hours of CC on the Machines of War PTR build 45889 (on the new map Braxis Holdout).

I also strongly believe Slows, Stuns, Roots, and Stasis should be separated.  Jaina often shows over 30 minutes of CC due to her using her AoE slows throughout the game, and Uther can finish a game with 8 seconds of stuns, though Uther's 8 seconds can be more impactful than all of Jaina's slows.

It would be more useful if I could compare Uther's stuns to Jaina's roots from her 'Numbing Blast' talent, and separate her standard AoE slows in a different field
